.github/workflows/ci.yml

```diff
@@ -411,3 +411,66 @@ jobs:
         run: >
             pytest
             -n 5 --maxfail 3 --durations 10 tests/unit tests/functional
+
+  test-cygwin:
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        include:
+          # Cygwin 3.4 dropped support for x86.
+          - { platform: x86_64 }
+      fail-fast: false
```

Comment:
Are you expecting to add more variants in the future? We don't need a matrix if there's only one job.

Reply:
Good point. Initially, I had `x86_64` and `x86` platforms in the matrix, but as per comment, it turned out that Cygwin 3.4 dropped support for `x86` (v3.4.0 was released in December 2022). Which in practice means that `x86` installation ends up with out-of-date `ca-certs` package, and `pip` cannot verify TLS connections. I suppose we could try temporarily disabling SSL to install up-to-date `certifi`, but I'm not sure how much demand there really is for x86 builds with Cygwin < 3.4.

Reply:
> I'm not sure how much demand there really is for x86 builds with Cygwin < 3.4.

23 out of about 150 CYGWIN downloads according to the PyPI statistics for the last 90 days. That's about a third of the number of nutters trying to run PyInstaller on their WiFi routers.

Reply:
Hmm, let's leave it for another day, then. I managed to get past the SSL errors by adding a step that re-runs `/etc/postinstall/ca-certificates.sh` (which seems to fail during initial setup), but now it looks like either `waf` or `gcc` or one of the test programs crashes when trying compile bootloader - both during explicit building step and later on when we try to install PyInstaller.

Based on statistics, we should be looking into setting up an OpenWRT CI pipeline before x86 Cygwin, anyway :)